### PR TITLE
Add optional AUTO_PROVISIONING env var

### PR DESCRIPTION
**Pull request by seanh**
Not used by anything yet. Example usage would be:

```python
if oauth_consumer_key in request.registry.settings["auto_provisioning"]:
    ...
```

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -112,6 +112,16 @@ The Hypothesis LMS app is written for python 3 and uses Node.js and `yarn` for m
     export H_CLIENT_SECRET="eVJ4***rXkk"
     export H_AUTHORITY="lms.hypothes.is"
     export H_API_URL="http://localhost:5000/api"
+
+    # An optional space-separated list of the consumer keys of the application
+    # instances for which the "auto provisioning" features should be enabled.
+    # If not given defaults to [] (i.e. the features are disabled for everyone).
```
Comment (lyzadanger) [inline README.markdown]: This sentence is a tad hard to parse without punctuation. Perhaps something like:

"Defaults to an empty list (`[]`) if none provided (i.e. the features will be disabled for all application instances)."

File: tests/lms/conftest.py [hunk 2]
```diff
@@ -150,6 +150,7 @@ def pyramid_config(pyramid_request):
         'h_client_secret': 'TEST_CLIENT_SECRET',
         'h_authority': 'TEST_AUTHORITY',
         'h_api_url': 'https://example.com/api',
+        'auto_provisioning': 'Hypothesise3f14c1f7e8c89f73cefacdd1d80d0ef Hypothesisf6f3a575c0c73e20ab41aa6be09b9c20',
```
Comment (lyzadanger) [inline tests/lms/conftest.py]: Any value for some sort of test where this is empty?

Comment (seanh) [reply tests/lms/conftest.py]: I've added a bunch of tests for various possibilities

File: README.markdown (hunk 1 above)
Comment (seanh) [reply README.markdown]: Done!
